model/task_lifecycle.go

```diff
@@ -616,7 +616,7 @@ func doBisectStepback(ctx context.Context, t *task.Task) error {
 	// last failing or last passing task.
 	if t.Status == evergreen.TaskSucceeded {
 		s.LastPassingStepbackTaskId = t.Id
-	} else if t.Status == evergreen.TaskFailed {
+	} else if evergreen.IsFailedTaskStatus(t.Status) {
```

Comment:
Maybe L1312 would change the logic, but I don't think this change would actually change how this logic actually runs. `(task.Task).Status` stores the DB model's status, which can only be `evergreen.TaskSucceeded` or `evergreen.TaskFailed` - all other statuses are display statuses, which have meaning in the UI but would never be stored in `(task.Task).Status`. There's [some discussion of this distinction between DB status vs. UI status here](https://github.com/evergreen-ci/evergreen/blob/59404295454cf10be3aead058fbe14fdf10444ce/globals.go#L45-L68) if you're interested.

In addition, it may be worth either:
1. Calling out in the doc comment on `evalStepback` that the `status` passed to `evalStepback` is the _display status_ (i.e. what the UI shows) rather than the DB task status (i.e. `(task.Task).Status`) or
2. If you don't have a specific need to check specific failure types like setup-failed/system-failed (I glanced at linear and bisect and don't see any indication that it needs to check the display status), you can just remove the `status` passed in and just use `(task.Task).Status` where necessary in linear stepback.

Reply:
That makes sense. I don't think either need specific statuses either. I'm going to try debugging more on how it failed for mms then

Reply:
I looked in to it a bit and stepback does use the display status in terms of activation (some [work](https://github.com/evergreen-ci/evergreen/pull/6062) Annie did). I think it makes sense to have bisect stepback activate on the same conditions, and I made it a bit more clear what is happening and why we need the display status